src/Configuration.php

```diff
@@ -155,4 +161,24 @@ public function setValidator(Validator $validator): void
     {
         $this->validator = $validator;
     }
+
+    public function getValidationConstraints(): array
+    {
+        if ($this->constraints === null) {
+            return [];
+        }
+
+        return $this->constraints;
```

Comment:
This would be simplified if we set the default value. Another way to simplify this method would be:

```php
public function getValidationConstraints(): array
{
    return $this->constraints ?? [];
}
```

I think I'd have the former but am open to discuss possibilities here :)

Reply:
Yes, it's simpler. I use here the guard clauses pattern sometimes it's more clear. https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html